packages/loader/container-loader/src/connectionManager.ts

```diff
@@ -1003,10 +1009,14 @@ export class ConnectionManager implements IConnectionManager {
 	 * @param error - Error reconnect information including whether or not to reconnect
 	 * @returns A promise that resolves when the connection is reestablished or we stop trying
 	 */
-	private reconnectOnError(requestedMode: ConnectionMode, error: IAnyDriverError): void {
-		this.reconnect(requestedMode, { text: error.message, error }).catch(
-			this.props.closeHandler,
-		);
+	private reconnectOnError(
+		requestedMode: ConnectionMode,
+		error: IAnyDriverError | undefined,
+	): void {
+		this.reconnect(requestedMode, {
+			text: error?.message ?? "Client closing delta connection",
```

Comment:
Here we are just guessing that it is "Client closing delta connection" and not something else.

Reply:
This is the string we are using currently for all disconnects, I did not introduce this string

Reply:
So, if the disconnection is from the socket side, we provide the error. For all other cases, where dispose is coming from the loader, as far as connection is concerned we are just saying that it is the client who is closing this conneciton. I think, we can just create an error type corresponding to the Client Closing delta connection and emit that on connection object and listener would then just check that error type and deduce why the connection get closed instead of undefined.